Update DID DHT Key Type to EdDSA

nitro-neal opened the pull request:
This bug fixes this issue and changes the index 0 key type from `Ed25519` to `EdDSA` (pending change coming to did-dht.com)

![image](https://github.com/TBD54566975/web5-js/assets/5314059/88a6dd4b-86ac-41cc-ac58-eb7bb756a488)
 
Here was the issue that was happening which this  pr fixes:

When creating a portable did with a jwk like this:
```
        privateKeys: [
          {
            crv : 'Ed25519',
            d   : 'hdSIwbQwVD-fNOVEgt-k3mMl44Ip1iPi58Ex6VDGxqY',
            kty : 'OKP',
            x   : 'VYKm2SCIV9Vz3BRy-v5R9GHz3EOJCPvZ1_gP1e3XiB0',
            kid : 'cyvOypa6k-4ffsRWcza37s5XVOh1kO9ICUeo1ZxHVM8',
            alg : 'EdDSA',
          }
        ]
```
        
(notice Alg is EdDSA)

the resolver gives:
```
    "verificationMethod": [
      {
        "id": "did:dht:ksbkpsjytbm7kh6hnt3xi91t6to98zndtrrxzsqz9y87m5qztyqo#0#0",
        "type": "JsonWebKey",
        "controller": "did:dht:ksbkpsjytbm7kh6hnt3xi91t6to98zndtrrxzsqz9y87m5qztyqo#0",
        "publicKeyJwk": {
          "kty": "OKP",
          "crv": "Ed25519",
          "x": "VYKm2SCIV9Vz3BRy-v5R9GHz3EOJCPvZ1_gP1e3XiB0",
          "kid": "cyvOypa6k-4ffsRWcza37s5XVOh1kO9ICUeo1ZxHVM8",
          "alg": "Ed25519"
        }
      }
    ],
```

(notice alg is Ed25519)

The solution was to change the actual did dht spec so that this does not happen and allows use to verify jwts with did dht

nitro-neal commented on pnpm-lock.yaml:
```diff
@@ -330,14 +330,14 @@ importers:
-        specifier: 1.0.0
-        version: 1.0.0
+        specifier: 1.0.1
+        version: link:../common
```
is this fine?

thehenrytsai commented on packages/credentials/tests/jwt.spec.ts:
```diff
@@ -72,6 +82,113 @@ describe('Jwt', () => {
+
+    it('successful verify with did:dht', async () => {
+      const hexString =
+        '0ab2b3386e22595e1271e7ef67fda70c37acf7d28b8c884a6fdcbb0ea739f341' +
```
Not a fan of opaque inputs to tests. Can you update and enable vector tests accordingly for the js implementation and call out in https://github.com/TBD54566975/web5-js/issues/638  that the kid changes must be reverted when 638 is fixed.

I think they should cover this change. Let me know if I am not making sense!

nitro-neal replied:
This test tests the Jwt.sign / verify with did dht, (the reason we had to change the spec a bit in the first place)

thehenrytsai replied:
Got it, assuming the main intent of this test is to test sign and verify, can we avoid using the opaque hex string in the test since it is besides the point?

nitro-neal commented:
I have turned the test vectors back on, they can pass by either:
1. what i've done here in the pr, changing the kid in the test vector
OR 
2. I can add this line to our fromDnsPacket parsing:
```js
          if(dnsRecordId === 'k0') {
            publicKey.kid = '0';
          }
```

![image](https://github.com/TBD54566975/web5-js/assets/5314059/1cbf4760-121c-4b34-ac9c-fa15e7eded6f)


Not sure which way is right at this point! but feel free to let me know which way we want to go

decentralgabe commented on packages/dids/tests/fixtures/test-vectors/did-dht/vector-2.json:
```diff
@@ -9,8 +9,8 @@
         "publicKeyJwk": {
-          "kid": "0",
-          "alg": "Ed25519",
+          "kid": "HP6bkG6mv-YPsU3Vi5Coi5wGYSVW9abFBwBSQwLQ7PU",
```
```suggestion
          "kid": "0",
```

decentralgabe commented on packages/dids/tests/fixtures/test-vectors/did-dht/vector-1.json:
```diff
@@ -7,8 +7,8 @@
         "publicKeyJwk": {
-          "kid": "0",
-          "alg": "Ed25519",
+          "kid": "HP6bkG6mv-YPsU3Vi5Coi5wGYSVW9abFBwBSQwLQ7PU",
```
```suggestion
          "kid": "0",
```

nitro-neal commented:
I have updated, 

added this line:
```
   if(dnsRecordId === 'k0') {
     publicKey.kid = '0';
   }
```

And chnaged the test vectors to have kid as 0 again

thehenrytsai commented on packages/dids/tests/methods/did-dht.spec.ts:
```diff
@@ -1298,7 +1298,7 @@ describe('Official DID:DHT Vector tests', () => {
   // Temporarily disabling due to inability to add a custom `kid` to our JWK in the current deployment.
   // We disable the test instead of changing the official specd test vector.
   // TODO: https://github.com/TBD54566975/web5-js/issues/638
```
Get rid of these, no longer applicable.

thehenrytsai commented on packages/dids/src/methods/did-dht.ts:
```diff
@@ -1055,6 +1055,10 @@ export class DidDhtDocument {
 
           publicKey.alg = parsedAlg || KeyTypeToDefaultAlgorithmMap[Number(t) as DidDhtRegisteredKeyType];
 
+          if(dnsRecordId === 'k0') {
+            publicKey.kid = '0';
+          }
+
```
Can't do this, else will break things. This is what @LiranCohen reverted. You should update the vector instead to use the thumbprint as `kid` temporarily. And optionally update #638 to mention what you changed so @LiranCohen can revert when #638 is done.

nitro-neal replied:
updated

thehenrytsai reviewed (approved):
🚀
